datalad_registry/tasks.py

```diff
@@ -70,6 +97,37 @@ def _extract_annex_info(repo) -> InfoType:
     return info
 
 
+@celery.task
+def collect_dataset_uuid(urls: Optional[List[str]] = None) -> None:
+    from flask import current_app
+
+    cache_dir = Path(current_app.config["DATALAD_REGISTRY_DATASET_CACHE"])
+    cache_dir.mkdir(parents=True, exist_ok=True)
+
+    if not urls:
+        lgr.debug("Did not find URLs that needed UUIDs collected")
+        return
+
+    lgr.info("Collecting UUIDs for %s URLs", len(urls))
+    lgr.debug("URLs: %s", urls)
+
+    for url in urls:
+        result = db.session.query(URL).filter_by(url=url)
+        if result.first() is not None:
+            result.update({"update_announced": 1})
+        else:
+            ds_path = cache_dir / "UNKNOWN" / url_encode(url)
+            ds = clone_dataset(url, ds_path)
+            ds_id = ds.id
```

Comment:
ds.id can be None if URL points to a regular git repo, which is without `.datalad/config` with the id. Then it would be None
```
$> python3 -c 'import datalad.api as dl; ds = dl.clone("https://github.com/datalad/datalad-registry", "/tmp/datalad-registry-clone"); print(ds.id)'
None 
```
In such case, we can still get info, but we should skip renaming and may be allow for  ds_id to be `None` in the model?  I guess change models to

```
ds_id = db.Column(db.Text, nullable=True)
```

?  Later PATCH could trigger re-checking of the repo and possibly assign UUID.

Reply:
Done.